tests/ui/resolve/issue-56593-2.rs

```diff
@@ -0,0 +1,26 @@
+// check-pass
+
+use thing::*;
+
+#[derive(Debug)]
+pub enum Thing {
+    Foo,
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_thing() {
+        let thing = Thing::Foo;
```

Comment:
```suggestion
        let thing: crate::Thing = Thing::Foo;
```
To make sure the types match.